sys/dev/nmdm/nmdm.c

```diff
@@ -115,16 +115,16 @@ nmdm_close(struct tty *tp)
 		return;
 
 	/* Shut down self. */
+	tty_lock(tp);
```

Comment:
This is wrong, `tp` is locked on entry

Reply:
Right: I mixed up the locks here, and this effective double locking (in the later commit where this lock is called from nmdm_destroy) hides a bug too. I'll make some changes and post them here.

Reply:
"Fixed" within the realm of the second and third patch.